pylint/lint/run.py

```diff
@@ -58,6 +58,13 @@ def _query_cpu() -> int | None:
             cpu_shares = int(file.read().rstrip())
         # For AWS, gives correct value * 1024.
         avail_cpu = int(cpu_shares / 1024)
+
+    # In K8s Pods also a fraction of a single core could be available
+    # As multiprocessing is not able to run only a "fraction" of process
+    # assume we have 1 CPU available
+    if avail_cpu == 0:
+        avail_cpu = 1
+
     return avail_cpu
```

Comment:
```suggestion
    return max(avail_cpu, 1)
```

A little bit more explicit imo. I didn't know `0,1 == 0` so people might forget that as well later on.

Reply:
``0,1 == 0 `` can't happen here, as it's getting rounded to the next ``int`` so ``avail_cpu`` is either ``0`` or ``>=1`` there.  Nevertheless, what I also thought of was, if just shortcircuiting the return wouldn't be the more elegant solution:

```suggestion
   return avail_cpu or 1
```

Reply:
I did apply your suggestion, but this is failing due to ``avail_cpu`` being initialized as ``None`` (a valid return value for this function imo) which does not work with ``max``. My proposal above also doesn't work here. If the function doesn't find any of the files used to determine cpus, it is just wrong to return ``1`` cpu available then.

Therefore I just reverse that change to my initial implementaion, only in case of ``avail_cpu == 0`` this should be set automatically to ``1``